### PR TITLE
Add Overlay for Infinix S4 (X626)

### PR DESCRIPTION
**Pull request by dmurimi14**
- **Infinix S4 Overlay** serves majorly to fix the notch configuration and automatic brightness on treble custom Roms.
- **Infinix ZERO 6 Overlay** fixes automatic brightness on custom Roms especially treble GSIs while at the same time bringing fixes for the notch configuration including proper handling of WhatsApp statuses display.

### CODE REVIEW

File: Infinix/S4/res/values/notch.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+-->
+<resources>
+    <!-- Height of the status bar in portrait -->
+    <dimen name="status_bar_height_portrait">50px</dimen>
+    <!-- Height of the status bar -->
+    <dimen name="status_bar_height">50px</dimen>
```
Comment (TadiT7) [inline Infinix/S4/res/values/notch.xml]: S4 doesn't need `status_bar_height`, it started with Pie. Tho overall, it's mergeable.

Comment (dmurimi14) [reply Infinix/S4/res/values/notch.xml]: I tried one GSI and there were issues with the status bar being pushed below the notch when using a higher dpi than the default one. This is what made me work on the overlay

Comment (dmurimi14) [reply Infinix/S4/res/values/notch.xml]: > S4 doesn't need `status_bar_height`, it started with Pie. Tho overall, it's mergeable.

I tried one GSI and there were issues with the status bar being pushed below the notch when using a higher dpi than the default one. This is what made me work on the overlay

Comment (phhusson) [reply Infinix/S4/res/values/notch.xml]: Thing is, even if I merge it, I'll remove the status_bar_height from pie branch anyway

Comment (dmurimi14) [reply Infinix/S4/res/values/notch.xml]: > Thing is, even if I merge it, I'll remove the status_bar_height from pie branch anyway

No problem sir, I understand

Comment (TadiT7) [reply Infinix/S4/res/values/notch.xml]: > > S4 doesn't need `status_bar_height`, it started with Pie. Tho overall, it's mergeable.
> 
> I tried one GSI and there were issues with the status bar being pushed below the notch when using a higher dpi than the default one. This is what made me work on the overlay

I meant just `status_bar_height`, not `status_bar_height_portrait`. No sane Pie ROM should use the former. BTW, if you meant that notch was bigger than the statusbar with "higher dpi" set, then it wasn't dpi what you changed, but dp display size in dev settings, which is basically reverse of dpi. The issue will be fixed either way.

Comment (dmurimi14) [reply Infinix/S4/res/values/notch.xml]: > > > S4 doesn't need `status_bar_height`, it started with Pie. Tho overall, it's mergeable.
> > 
> > 
> > I tried one GSI and there were issues with the status bar being pushed below the notch when using a higher dpi than the default one. This is what made me work on the overlay
> 
> I meant just `status_bar_height`, not `status_bar_height_portrait`. No sane Pie ROM should use the former. BTW, if you meant that notch was bigger than the statusbar with "higher dpi" set, then it wasn't dpi what you changed, but dp display size in dev settings, which is basically reverse of dpi. The issue will be fixed either way.

I get you know

**Pull request merged**